backend/bytecode.proto

```diff
@@ -0,0 +1,206 @@
+        // Magic number? Having one doesn't make sense if we're using protobuf
+        // since it can do validation of bytestreams.
+        required string major = 1;
+        required string minor = 2;
+    }
+    // i
+    message Import {
+        enum ImportType {
+            METADATA = 1;
+            TYPE = 2;
+        }
+        optional ImportType importType = 1;
+        required string platform = 2;
+        // Need to make a global decision here: In the bytecode should we store
+        // paths as '.' delimited strings or arrays of strings? What about
+        // little 'p' paths used in match expressions, datatags, etc?
+        required string path = 3;
+        required Type type = 4;
+        required string name = 5;
+    }
+    required Version version = 1;
+    required string path = 2;
+    repeated Import imports = 3;
+    required Module module = 4;
+}
+
+// M
+message Module {
+    message ExpressionModule {
+        required string path = 1;
+        required Type type = 2;
+        required Expression expression = 3;
+    }
+    oneof module {
+        ExpressionModule expressionModule = 1;
+        TypeDeclaration typeModule = 2;
+    }
+}
+
+// e
+message Expression {
+    message NewExpression {
+        required Type type = 1;
+        required string this = 2;
+        repeated Declaration declarations = 3;
+    }
+    message CallExpression {
+        required Expression callee = 1;
+        required string method = 2;
+        repeated Expression arguments = 3;
+    }
+    message AccessExpression {
+        required Expression expression = 1;
+        required string field = 2;
+    }
+    message AssignmentExpression {
+        required Expression expression = 1;
+        required string field = 2;
+        required Expression value = 3;
+    }
+    message LetExpression {
+        required string variable = 1;
+        required Expression value = 2;
+        required Expression expression = 3;
+    }
+    message Literal {
+        oneof literal {
+            string string = 1;
+            bytes integer = 2;
+        }
+    }
+    message MatchExpression {
+        message MatchArm {
+            required string variable = 1;
+            required string path = 2;
+            required Expression expression = 3;
+        }
+        required Expression expression = 1;
+        repeated MatchArm arms = 2;
+        optional Expression else = 3;
+    }
+    oneof expression {
+        string variable = 1;
+        NewExpression newExpression = 2;
+        CallExpression callExpression = 3;
+        AccessExpression accessExpression = 4;
+        AssignmentExpression assignmentExpression = 5;
+        LetExpression letExpression = 6;
+        Literal literal = 7;
+    }
+}
+
+// d
+message Declaration {
+    message VariableDeclaration {
+        required VariableDeclarationType declarationType = 1;
+        required string variable = 2;
+        required Type type = 3;
+        required string value = 4;
```

Comment:
Let's make this an expression instead of a value.  The actual IL used in Wyvern allows this, even if the VM spec document doesn't.  Call it the initializer.